rules/library.bzl

```diff
@@ -639,36 +640,29 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
         import_vfsoverlays.append(import_name + "_vfs")
 
     for vendored_static_library in kwargs.pop("vendored_static_libraries", []):
+        # Collect the include paths which are the top-level directories for where the headers are.
+        # Example: "Headers/ObjcLib/Foo.h" -> includes = ["Headers"]
+        includes = collections.uniq([hdr.split("/")[0] for hdr in objc_hdrs])
         import_name = "%s-%s-library-import" % (name, paths.basename(vendored_static_library))
         native.objc_import(
             name = import_name,
             archives = [vendored_static_library],
+            hdrs = objc_hdrs,
+            includes = includes,
```

Comment:
We don't support `include` in rules_ios - because of all the issues it brings to the table and we have optimizations banking that it isn't used. Do you have an example where this fixes anything? Adding this line looks like a symptom of some other issue

Reply:
Hmm - also in practice this might not be as big of an issue assuming it is correctly stripped off elsewhere

Reply:
The fixture I added in this repository doesn't build without the `includes` & `hdrs` change 
in `objc_import` and if `public_headers` is empty. Which means that the code as it is today in `master` wasn't working as intended (i.e. collecting headers implicitly). 

The main issue this PR and #530 attempt to fix, comes from `apple_library` doing a bunch of stuff (creating hmap, objc_library, etc) based on the `objc_hdrs`. This property is empty when `public_headers` is empty. Which is the case when the `BUILD` file doesn't define `public_headers`, it will also be empty for `vendored_xcframeworks` since headers are in multiple slices and passing a `select` doesn't work with the code as it is today.

I think the original intent of the code as it is today was to separate the vendored libraries and the top level target that defines them into separate targets. For example, create different `framework_vfs_overlay` for the library and the top-level target. 

So my attempt at fixing this was correcting usage of `objc_import` since in the Bazel docs it says using `includes` allows the `#import <>` statements to work without changing too much of the existing code (implicit collection of headers, etc).

If we want to not provide `includes` to `objc_import` then we need to revert #530 as well and rework `apple_library` to somehow support `select` for `objc_hdrs` (since that will be required for the different slices). This will also change the rest of the code since we implicitly collect headers for every single vendored library (see `import_headers`).

This has all been from me trying to understand the original code, so if I'm missing something then please correct me but it was difficult enough to understand where everything was being created and how these things are intended to work since we never had tests for this in the repo.